test/module/webpage/load-images-no.js

```diff
@@ -0,0 +1,52 @@
+//! phantomjs: --load-images=no
+
+// Compiled for debug, this tests for leaks when not loading images.
+//
+
+var url = TEST_HTTP_BASE + 'load-images.html';
+
+function do_test() {
+    var page = require('webpage').create();
+
+    var numImagesRequested = 0;
+    var numImagesLoaded = 0;
+    
+    page.onResourceRequested = this.step_func(function (resource) {
+        if (/\.png$/.test(resource.url)) ++numImagesRequested;
+    });
+
+    page.onResourceReceived = this.step_func(function (resource) {
+        if (resource.stage === 'end' && /\.png$/.test(resource.url)) ++numImagesLoaded;
+    });
+
+    function checkImageCounts ()
+    {
+        var imageCounts = page.evaluate (function() {    
+            "use strict";
+            try
+            {
+                var images = document.getElementsByTagName('img');
+                var imageCounts = {};
+                imageCounts.numImages = images.length;
+                imageCounts.numComplete = Array.prototype.reduce.call(images, function(n, image) { return n + (image.complete? 1 : 0); }, 0);
+                return imageCounts;
+            }
+            catch(err)
+            {
+                return undefined;
```

Comment:
Logging inside evaluate won't surface unless you use onConsoleMessage.

I played around with what happens when I intentionally throw an exception. In this case, it's not the return value that gets checked, but returnvalue.someField. This results in e.g.

	module/webpage/load-images-yes: FAIL
	 FAIL: load a page two images with --load-images=yes
	  null is not an object (evaluating 'imageCounts.numImages')
	  checkImageCounts@phantomjs://code/load-images-yes.js:38:34
	  pageOpened@phantomjs://code/load-images-yes.js:46:25

If I remove the try/catch from evaluate, the exception does get reported, but it's pretty strange:

	module/webpage/load-images-yes: ERROR
	ERROR: neither a test nor a diagnostic
	  Error: oops
	 FAIL: test 1 did not report status

Error: oops is the exception I added, but in most cases you wouldn't see the explicit throw in the test code. The other way seems clearer to me.